test.py

```diff
@@ -202,11 +210,60 @@ def test_sample_without_suffix(self):
     def tearDownClass(cls):
         os.unlink(cls.test_db)
 
+class TestRules(unittest.TestCase):
+    """
+    Unittests for Rules.
+
+    @author: Felix Bauer
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls.conf = PeekabooDummyConfig()
+        cls.factory = SampleFactory(cuckoo = None,
+                db_con = None,
+                connection_map = None,
+                base_dir = cls.conf.sample_base_dir,
+                job_hash_regex = cls.conf.job_hash_regex,
+                keep_mail_data = False)
+        cls.sample = cls.factory.make_sample(os.path.realpath(__file__))
+
+    def test_rule_file_type_on_whitelist(self):
+        combinations = [
+            [False, ['text/plain']],
+            [True, ['application/vnd.ms-excel']],
+            [True, ['text/plain', 'application/vnd.ms-excel']],
+            [True, ['image/png', 'application/zip', 'application/vnd.ms-excel']],
+            [True, ['', 'asdfjkl', '93219843298']],
+            [False, []], # should this really result in no further_analysis?
+        ]
+        for expected, types in combinations:
+            self.sample.set_attr('mimetypes', set(types))
+            r = file_type_on_whitelist(self.conf, self.sample)
+            self.assertEqual(r.further_analysis, expected)
+
+    def test_rule_file_type_on_greylist(self):
+        combinations = [
+            [False, ['text/plain']],
+            [True, ['application/msword']],
+            [True, ['text/plain', 'application/x-dosexec']],
+            [True, ['image/png', 'application/zip', 'application/vnd.ms-excel', 'application/vnd.ms-powerpoint']],
+            [False, ['', 'asdfjkl', '93219843298']],
+            [False, []],
```

Comment:
How about this one: As said, shouldn't not knowing any MIME type for a sample also cause it to be greylisted, i.e. being analysed? Otherwise the above correction to the whitelisting logic is moot.

Reply:
Exactly. Applies here also

Reply:
Pefect.